trackintel/preprocessing/positionfixes.py

```diff
@@ -164,7 +159,11 @@ def generate_staypoints(
     # user_id of sp should be the same as ret_pfs
     sp["user_id"] = sp["user_id"].astype(pfs["user_id"].dtype)
 
-    return pfs, sp
+    if len(sp) <= 0:
+        warnings.warn("No staypoints can be generated, returning empty sp.")
+        return pfs, sp
```

Comment:
Do we need this line? Can we also use `Staypoints(sp)` even if `sp` is empty? This relates to #551.

Reply:
Yes, `Staypoints.validate` will fail when it tries to access the first element for the geometry type.